src/devices/usbCamera/CMakeLists.txt

```diff
@@ -4,19 +4,60 @@
 # This software may be modified and distributed under the terms of the
 # BSD-3-Clause license. See the accompanying LICENSE file for details.
 
+include(FetchContent)
+
+option(COMPILE_WITHUNITTEST_ULTRAPYTHON "Compile also unittest for UltraPython cam board" OFF)
+
+# Get rid of undef while compiling gtest -- START
+get_property(
+    compile_options
+    DIRECTORY
+    PROPERTY COMPILE_OPTIONS
+)
+set_property(
+    DIRECTORY
+    APPEND
+    PROPERTY COMPILE_OPTIONS -Wno-undef -Wno-error
+)
+
+
+# Get rid of undef while compiling gtest -- END
+
+
+#test
+if(COMPILE_WITHUNITTEST_ULTRAPYTHON)
+  FetchContent_Declare(
+      googletest
+      GIT_REPOSITORY "https://github.com/google/googletest"
+      GIT_TAG        master
+  )
+  set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+  FetchContent_MakeAvailable(googletest)
+  add_subdirectory(unittest)
+  endif()
+
+# Get rid of undef while compiling gtest -- START
+set_property(
+    DIRECTORY
+    PROPERTY COMPILE_OPTIONS ${compile_options}
+)
+unset(compile_options)
```

Comment:
This part of gtest has to be removed, see the comment below about the use of catch